memory.cc

```diff
@@ -61,9 +61,9 @@ std::vector<float> PerfectMemory::calculateRIDF(const cv::Mat &snapshot) const
     std::vector<float> ridf(getImageSize().width, std::numeric_limits<float>::max());
 
     // Loop through all snapshots
-    for(const auto &memDifferences : allDifferences) {
-        for(int c = 0; c < getImageSize().width; c++) {
-            ridf[c] = std::min(ridf[c], memDifferences[c]);
+    for (int i = 0; i < allDifferences.rows(); i++) {
```

Comment:
is this not a vector of vectors anymore?

Reply:
Nope, I changed it to be an Eigen matrix instead. We're using Eigen for some of the other navigation code anyway, so I thought we might as well use it here. It just makes it easier in terms of resizing etc. and I think it's a tiny bit faster too...

Reply:
👍